### ISSUE TITLE
📎 React 19.2: add useEffectEvent semantics to useExhaustiveDependencies and recognise <Activity />

### ISSUE THREAD

**Issue by barklund** [biomejs/biome#7631]
### Description

[React 19.2](https://react.dev/blog/2025/10/01/react-19-2) adds two built-ins that Biome should understand:

### Summary
React 19.2 introduces two new built-ins that Biome should support:  
- [The new built-in hook](https://react.dev/blog/2025/10/01/react-19-2#use-effect-event) `useEffectEvent` has a stable, ref-like return value that **must not** be added to dependency arrays (should be treated like [`useRef`](https://github.com/biomejs/biome/blob/fa4fcd355f8bce46b83a60fc2a874bfa93c2ce4f/crates/biome_js_analyze/src/lint/correctness/use_exhaustive_dependencies.rs#L293-L306)).  
- [The new built-in component](https://react.dev/blog/2025/10/01/react-19-2#activity) `<Activity />` should be added to the list of [known React API elements](https://github.com/biomejs/biome/blob/fa4fcd355f8bce46b83a60fc2a874bfa93c2ce4f/crates/biome_js_analyze/src/react.rs#L140).

### Why
- Align Biome’s `useExhaustiveDependencies` rule with React 19.2 semantics so developers don’t get false positives when using Effect Events.  
- Keep Biome’s React API list up to date by recognising new built-ins introduced in React releases.

### Proposed changes
- In [`use_exhaustive_dependencies.rs`](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/lint/correctness/use_exhaustive_dependencies.rs#L293-L306), treat `useEffectEvent` like `useRef`:  
  - Calls to `useEffectEvent` should be recognised as producing a stable, ref-like value that is **never** added to dependency arrays.  
  - Example:  
    ```tsx
    const onConnected = useEffectEvent(() => { ... });
    useEffect(() => { ... }, []); // should not warn about missing `onConnected`
    ```
  - Update the test suite by adding representative cases to [valid.js](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/tests/specs/correctness/useExhaustiveDependencies/valid.js) (no missing dep warning for Effect Events) and [missingDependenciesInvalid.jsx](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/tests/specs/correctness/useExhaustiveDependencies/missingDependenciesInvalid.jsx#L11) (ensure adding an Effect Event is flagged as unnecessary).

- In [`react.rs`](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/react.rs#L140), add:  
  - `useEffectEvent` to the known React Hooks list.  
  - `Activity` to the known React Components list.  
  - Example:  
    ```tsx
    import { Activity } from "react";
    <Activity />; // should be recognised as a React API element
    ```

**Comment by aseerkt** [comment]
this PR can be closed. @barklund
